src/components/ebay-breadcrumb/test/test.server.js

```diff
@@ -0,0 +1,68 @@
+const expect = require('chai').expect;
+const testUtils = require('../../../common/test-utils/server');
+const mock = require('../mock/index');
+
+describe('should render with basic breadcrumb and ', () => {
+    test('return nav element with input', context => {
+        const $ = testUtils.getCheerio(context.render(mock.basicItems));
+        const nav = $('nav');
+        expect(nav.length).to.equal(1);
+        expect(nav.attr('role')).to.equal('navigation');
+        expect(nav.attr('dataview')).to.equal('data-view tracking');
+        expect(nav.attr('class')).to.equal('breadcrumb');
+        expect(nav.attr('aria-labelledby')).to.equal('breadcrumbs_heading');
+    });
+    test('return h2 tag', context => {
+        const $ = testUtils.getCheerio(context.render(mock.basicItems));
+        const h2Tag = $('h2');
+        expect(h2Tag.length).to.equal(1);
+        expect(h2Tag.attr('id')).to.equal('breadcrumbs_heading');
+        expect(h2Tag.attr('class')).to.equal('clipped');
+        expect(h2Tag.html()).to.equal('Page navigation');
+    });
+    test('return items html attributes', context => {
+        const $ = testUtils.getCheerio(context.render(mock.basicItems));
+        const anchor = $('nav li a');
+        const _sp = [];
+        const navsrc = [];
+        anchor.each((i, v) => {
+            _sp.push($(v).attr('_sp'));
+            navsrc.push($(v).attr('navsrc'));
+        });
+        expect(_sp.length).to.equal(4);
+        expect(navsrc.length).to.equal(4);
+    });
+    test('create <a> tag if href is NOT null for last item', context => {
+        const $ = testUtils.getCheerio(context.render(mock.basicItems));
+        const li = $('nav li');
+        expect(li.length).to.equal(4);
+        const anchor = $('nav li a');
+        expect(anchor.length).to.equal(4);
+    });
+});
+
+test('should return zero nav element in the page', context => {
+    const input = {};
+    const $ = testUtils.getCheerio(context.render(input));
+    expect($('nav').length).to.equal(0);
+});
+test('should create <a> without href if href attr is passed null unless its last item', context => {
+    const $ = testUtils.getCheerio(context.render(mock.itemsWithMissingLinks));
+    const li = $('nav li');
+    expect($('a', li[0]).attr('href')).to.be.an('undefined');
+    expect(li.length).to.equal(4);
+});
+test('should create span tag if href is null ', context => {
+    const $ = testUtils.getCheerio(context.render(mock.itemsWithNoHref));
+    const li = $('nav li');
+    expect(li.length).to.equal(4);
+    const currentElement = $('span', li[li.length - 1]);
+    expect(currentElement.attr('aria-current')).to.equal('page');
+});
+test('return headingLevel', context => {
+    const $ = testUtils.getCheerio(context.render(mock.itemsWithHeadingLevel));
+    const h2Tag = $('h2');
+    expect(h2Tag.length).to.equal(0);
+    const h3Tag = $('h3');
+    expect(h3Tag.length).to.equal(1);
+});
```

Comment:
Can you use the common utils to test pass-through on both the root and the `li`s?
https://github.com/eBay/ebayui-core/blob/fa69045f03d59b5c5f9b749645bb8af32f6dde37/src/components/ebay-menu/test/test.server.js#L47-L53

Reply:
Tried with testUtils.testHtmlAttributes(context, 'nav li a') -- looks like its rendering with * attr input. it does not work as for me. can it be generalized?

Reply:
The util can be changed if needed, but I think it should work as-is for your case. Take a look at the 2 examples in the menu case, where it calls the util slightly differently for the cases of root vs. nested tag.

Reply:
this test is added at L30-L36